src/library/scala/annotation/inheritSignature.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/**
+ * Methods annotated `@inheritSignature` are emitted `private` in bytecode, which
+ * allows adding binary compatible overrides when the bytecode signature is refined.
+ *
+ * Example:
+ *
+ * {{{
+ *   class A { def f: AnyRef = "A.f" }
+ *   class C extends A
+ * }}}
+ *
+ * Adding an override of `f` to `C` and refining the result type is not forwards
+ * binary compatible:
+ *
+ * {{{
+ *    class C extends A { override def f: String = "C.f" }
+ * }}}
+ *
+ * The class `C` gets a new method in bytecode with return type `String`. In order
+ * to implement overriding, the compiler generates a "bridge" method in `C` which
+ * has the same signature as the overridden method (return type `Object`).
+ *
+ * By annotating the override `@inheritSignature`, the new method `C.f: String`
+ * is made `private`, only the bridge method remains `public`.
+ *
+ * Invocations of `C.f: String` are rewritten to `A.f`.
```

Comment:
Testing this out:

```
scala> class A { def f: AnyRef = "A.f" }
class A

scala> class C extends A { override def f: String = "C.f" }
class C

scala> class D extends A { @annotation.inheritSignature override def f: String = "C.f"; def use = f }
class D


scala> :javap -private -v C#f
  public java.lang.String f();
    descriptor: ()Ljava/lang/String;
    flags: ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: ldc           #22                 // String C.f
         2: areturn

  public java.lang.Object f();
    descriptor: ()Ljava/lang/Object;
    flags: ACC_PUBLIC, ACC_BRIDGE, ACC_SYNTHETIC
    Code:
      stack=1, locals=1, args_size=1
         0: aload_0
         1: invokevirtual #31                 // Method f:()Ljava/lang/String;
         4: areturn

scala> :javap -private -v D#use
  public java.lang.String use();
    descriptor: ()Ljava/lang/String;
    flags: ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: aload_0
         1: invokevirtual #28                 // Method f:()Ljava/lang/Object;
         4: checkcast     #30                 // class java/lang/String
         7: areturn
```

Reply:
So looks good, no?

There is one issue in the repl:

```
scala> class A { def f: Object = "a" }
     | class B extends A { @annotation.inheritSignature override def f: String = "b" }

scala> val s: String = (new B).f
                               ^
       error: type mismatch;
        found   : Object
        required: String
```

It compiles fine in the batch compiler.

Reply:
> So looks good, no?

Yep, I was just documenting my testing, not highlighting an issue.